cpp/unit/ability.cpp

```diff
@@ -1,88 +1,242 @@
 // Copyright 2014-2015 the openage authors. See copying.md for legal info.
 
-#include "ability.h"
-
 #include <memory>
 
 #include "../terrain/terrain_object.h"
+#include "../datamanager.h"
+#include "ability.h"
 #include "action.h"
+#include "command.h"
 #include "unit.h"
 
 namespace openage {
 
-MoveAbility::MoveAbility(Texture *t, TestSound *s)
+std::string to_string(const ability_type &at) {
```

Comment:
we could place this function in the `namespace std` and provide specialized implementations for our types, e.g. the `ability_type`. ack?